dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -162,3 +162,161 @@ def submit(self, compiled_code: str) -> None:
 class ServerlessClusterPythonJobHelper(BaseDatabricksHelper):
     def submit(self, compiled_code: str) -> None:
         self._submit_through_notebook(compiled_code, {})
+
+
+class WorkflowPythonJobHelper(BaseDatabricksHelper):
+
+    @property
+    def default_job_name(self) -> str:
+        return f"{self.database}-{self.schema}-{self.identifier}__dbt"
+
+    @property
+    def notebook_path(self) -> str:
+        return f"{self.notebook_dir}/{self.identifier}"
+
+    @property
+    def notebook_dir(self) -> str:
+        return f"/Shared/dbt_python_model/{self.database}/{self.schema}"
+
+    def check_credentials(self) -> None:
+        workflow_config = self.parsed_model["config"].get("workflow_job_config", None)
+        if not workflow_config:
+            raise ValueError(
+                "workflow_job_config is required for the `workflow_job_config` submission method."
+            )
+
+    def submit(self, compiled_code: str) -> None:
+        workflow_spec = self.parsed_model["config"]["workflow_job_config"]
+        cluster_spec = self.parsed_model["config"].get("job_cluster_config", None)
+
+        # This dict gets modified throughout. Settings added through dbt are popped off
+        # before the spec is sent to the Databricks API
+        workflow_spec = self._build_job_spec(workflow_spec, cluster_spec)
+
+        self._submit_through_workflow(compiled_code, workflow_spec)
+
+    def _build_job_spec(self, workflow_spec, cluster_spec):
+        workflow_spec["name"] = workflow_spec.get('name', self.default_job_name)
+
+        cluster_settings = {}  # Undefined cluster settings defaults to serverless in the Databricks API
+        if cluster_spec is not None:
+            cluster_settings["new_cluster"] = cluster_spec
+        elif 'existing_cluster_id' in workflow_spec:
+            cluster_settings['existing_cluster_id'] = workflow_spec['existing_cluster_id']
+
+        notebook_task = {
+            'task_key': 'task_a',
+            'notebook_task': {
+                "notebook_path": self.notebook_path,
+                "source": "WORKSPACE",
+            },
+        }
+        notebook_task.update(cluster_settings)
+        notebook_task.update(workflow_spec.pop("additional_task_settings", {}))
+
+        post_hook_tasks = workflow_spec.pop("post_hook_tasks", [])
+        for task in post_hook_tasks:
+            if not 'existing_cluster_id' in task and not 'new_cluster' in task:
+                task.update(cluster_settings)
+
+        workflow_spec["tasks"] = [notebook_task] + post_hook_tasks
+        return workflow_spec
+
+    def _submit_through_workflow(self, compiled_code: str, workflow_spec) -> None:
+        self.api_client.workspace.upload_notebook(self.notebook_path, compiled_code)
+
+        job_id, is_new = self._get_or_create_job(workflow_spec)
+
+        if not is_new:
+            self.api_client.workflows.update_by_reset(job_id, workflow_spec)
```

Comment:
Even though the API is 'reset', I think update_job_settings is probably more apt.

Reply:
Yeah, I like that

Reply:
changed